data/postsRouter.js

```diff
@@ -0,0 +1,82 @@
+const router = require("express").Router();
+
+const db = require("./db.js");
+
+//-------- New Post  --------//
+router.post("/", (req, res) => {
+  const newPost = res.body;
+  if (newPost.title && newPost.contents) {
+    db.insert(newPost)
+      .then(post => {
+        res.status(201).json(newPost);
+      })
+      .catch(err => {
+        res.status(501).json({ message: "Failed to add Post" });
+      });
+  } else {
+    res.status(400).json({
+      Message: "Please provide title and contents for the post."
+    });
+  }
+});
+
+//------Get all Posts-------//
+router.get("/", (req, res) => {
+  db.find()
+    .then(posts => {
+      res.status(202).json(posts);
+    })
+    .catch(err => {
+      res.status(501).json({ message: "Failed To Get Posts" });
+    });
+});
+
+//-------- Get Posts By Id--------//
+
+router.get("/:id", (req, res) => {
+  db.findById(req.params.id)
+    .then(post => {
+      res.status(200).json(post);
+    })
+    .catch(err => {
+      res.status(407).json({ message: "That user doesn't exist" });
+    });
+});
+
+// // --------------------------------- DELETE -----------------------------------//
+router.delete("/:id", (req, res) => {
+  db.remove(req.params.id)
+    .then(remove => {
+      res.status(204).end();
```

Comment:
```suggestion
      res.status(204).end();
```
.end doesn't send anything in the response so please put useful information in here with .json such as a message saying deleting user was successful